processor/batchprocessor/batch_processor.go

```diff
@@ -286,6 +397,63 @@ func (b *shard) sendItems(trigger trigger) {
 	b.processor.telemetry.record(trigger, int64(sent), int64(bytes))
 }
 
+// parentSpans computes the set of distinct span contexts and returns
+// the corresponding set of active span objects.
+func parentSpans(x []pendingItem) []trace.Span {
```

Comment:
An SDK that is producing a lot of span/logs data, for example, could send multiple batches concurrently. If the collector batch processor is using larger batches than the SDK, then high-throughput SDKs are likely to have multiple of their small batches inside a collector's larger batch.

The https://github.com/lightstep/otel-launcher-go SDK which we use has been modified to use OTel-Collector processor and exporter components for spans and metrics, and we use the concurrent batch processor to enable higher throughput. It is this setup that leads to duplicates.

Besides, I'm not sure if you mean inefficient or unnecessary. This is an efficient algorithm to de-duplicate spans, though it could be unnecessary. I can be convinced to simply span-link them all and allow duplicates through.

Reply:
I see. Yes, something is wrong here.

Reply:
OK, here's what I did.

I removed the allSameContext() function and instead test `len(thisBatch) == 1` i.e., the special case is now for the case where a single item represents a full batch, which will be the case when one batching collector sends to another batching collector configured for similar-size batches.

Then, in the other branch, I don't compute unique spans -- just link them all. There is no chance of duplicates in a standard collector configuration, it was my oddball SDK setup that created the same context as parent.